hibernate-core/src/main/java/org/hibernate/dialect/Dialect.java

```diff
@@ -1484,11 +1484,29 @@ public String castPattern(CastType from, CastType to) {
 	 *
 	 * @param specification {@code leading} or {@code trailing}
 	 * @param character the character to trim
+	 *
+	 * @deprecated Use {@link #trimPattern(TrimSpec, boolean)} instead.
 	 */
+	@Deprecated( since = "6.5" )
 	public String trimPattern(TrimSpec specification, char character) {
-		return character == ' '
-				? "trim(" + specification + " from ?1)"
-				: "trim(" + specification + " '" + character + "' from ?1)";
+		return trimPattern( specification, character != ' ' );
+	}
+
+	/**
+	 * Obtain a pattern for the SQL equivalent to a
+	 * {@code trim()} function call. The resulting
+	 * pattern must contain a ?1 placeholder for the
+	 * argument of type {@link String} and a ?2 placeholder
+	 * for the trim character if {@code explicitCharacter}
+	 * was specified.
+	 *
+	 * @param specification {@linkplain TrimSpec#LEADING leading}, {@linkplain TrimSpec#TRAILING trailing}
+	 * or {@linkplain TrimSpec#BOTH both}
+	 * @param explicitCharacter {@code true} if the pattern must include a ?2 placeholder
+	 * for an explicit trim character, {@code false} otherwise
+	 */
+	public String trimPattern(TrimSpec specification, boolean explicitCharacter) {
```

Comment:
I was a bit confused when I first saw the changes due to the naming. What do you think about passing `boolean isWhitespace` instead and pass `false` if a parameter is given?

Reply:
Yeah I wasn't sure about the naming but couldn't decide on what to use. Thanks for the suggestion, I've changed to that.